core/src/datayoga_core/write_utils.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from itertools import groupby
+from typing import Any, Callable, Dict, List, Union
+
+from datayoga_core.block import Block
+from datayoga_core.opcode import OpCode
+from datayoga_core.result import Result, Status
+
+logger = logging.getLogger("dy")
+
+
+def write_records(data: List[Dict[str, Any]],
+                  opcode_field: str, execute_upsert: Callable, execute_delete: Callable) -> List[Dict[str, Any]]:
+    # group records by opcode
+    opcodes = [{"opcode": key, "records": list(result)} for key, result in groupby(
+        data, key=lambda record: record.get(opcode_field, "").replace(OpCode.CREATE.value, OpCode.UPDATE.value))]
+
+    for records_by_opcode in opcodes:
+        opcode = records_by_opcode["opcode"]
+        records: List[Dict[str, Any]] = records_by_opcode["records"]
+
+        logger.debug(f"Total {len(records)} record(s) with {opcode} opcode")
+        if opcode == OpCode.UPDATE.value:
+            execute_upsert(records)
+        elif opcode == OpCode.DELETE.value:
+            execute_delete(records)
+        else:
+            for record in records:
+                record[Block.RESULT_FIELD] = Result(Status.REJECTED, f"{opcode} - unsupported opcode")
+            logger.warning(f"{opcode} - unsupported opcode")
+
+    return data
+
+
+def get_source_column(item: Union[Dict[str, str], str]) -> str:
```

Comment:
what do these methods do? Do we really need them as a utility method? 
in any case, I would have clear comments since it is hard to follow what is their purpose

Reply:
they take an object or a string of column mapping, for example:
{ "last_name": "lname" }
or 
"last_name"

And return `last_name` for source and `lname"`for target.

Since we use them in a several places in `cassandra.write` and `relational.write`, I thought it would be good to have them reusable.

Reply:
removed them completely.